ExecutionOrder.md

```diff
@@ -0,0 +1,61 @@
+|internal/server/iigointernal/base_speaker.go|appointNextJudge|Subtract 10 as cost of action. Propose and carry out election of next judge and returns results. * |
+|internal/server/iifo.go|runIIFO|Runs prediction session. |
+|internal/server/iifo.go|runPredictionSession| Get islands' predictions and distribute predictions to all the islands. |
+|internal/server/iifo.go|getPredictions|For alive clients, get their prediction for disaster (and foraging - non-MVP) using MakePrediction which is implemented by each client.|
+|internal/server/iifo.go|distributePreictions|Share predictions with islands of your choice. Islands who receive a prediction(s) do not know who else receieved the same prediciton(s).|
+|internal/server/iito.go|runIITO|MVP: gifts session only.|
+|internal/server/iito.go|runGiftSession|Gets all fiftr requests and offers. Gets the result (accept/reject gift(s)) and distributes gifts accordingly. |
+|internal/server/iito.go|getGiftAcceptance|Matches gift requests and offers per client, then allows clients to accept/reject offers. Then, returns result (accept/reject) of offer/request pairs. |
+|internal/server/turn.go|endOfTurn|Runs the organisations end of turn, forage, rpobe for disasters, increase turn and season count, deduct cost of living from clients and check clients' alive/dead status. |
+|internal/server/turn.go|runOrgsEndOfTurn|Runs end of turn for IIGO then IIFO then IITO.|
+|internal/server/iigo.go|runIIGOEndOfTurn|Get tax from each alive client and add it to the common pool. Update the game state. Logs how much tax each client contributed. |
+|internal/server/iifo.go|runIIFOEndOfTurn| To be implemented. |
+|internal/server/iito.go|runIITOEndOfTurn| To be implemented. |
+|internal/server/forage.go|runForage|Calls getForagingDecisions (this calls DecideForage which is to be implemented by the client) to get foraging participants. MVP: deer hunting only via runDeerHunt.|
+|internal/server/forage.go|runDeerHunt|Creates a deer hunt for the participants and returns amount generated by foraging.|
+|internal/server/disaster.go|probeDisaster|Checks if a disaster occurs this turn. |
+|internal/server/turn.go|incrementTurnAndSeason|Increments turn. Increments season if a disaster occurs.|
+|internal/server/turn.go|deductCostOfLiving|Deduct cost of living from each alive client then update the gamestate. |
+|internal/server/turn.go|updateIslandLivingStatus|Update island status to dead is needed. |
+
+\* The same process is repeated for the other roles .
+
+## Questions
+
+- internal/server/iigointernal/orchestration.go - what happens if president and judge can be paid but speaker can't be? How are the salaries that are taken from the pool and stored returned back to the common pool? IIGO is exited before they can do their job.
```

Comment:
@Neelesh99 Correct me if im wrong, but if any of the three roles cannot be paid then all salaries are given back to the common pool in the returnWithdrawnSalariesToCommonPool function.